src/use-cases/authUserUseCase.ts

```diff
@@ -0,0 +1,40 @@
+import { User } from '@prisma/client'
+import { UserRepository } from '../repositories/user-repository'
+import { ResourceNotFoundError } from './errors/ResourceNotFoundError'
+import { compare } from 'bcryptjs'
+
+interface AuthUserUseCaseRequest {
+  email: string
+  password: string
+}
+
+interface AuthUserProfileUseCaseResponse {
+  user: User
+}
+
+export class AuthUserUseCase {
+  constructor(private userRepository: UserRepository) {}
+
+  async execute({
+    email,
+    password,
+  }: AuthUserUseCaseRequest): Promise<AuthUserProfileUseCaseResponse> {
+    const user = await this.userRepository.findByEmail(email)
+
+    if (!user) {
+      throw new ResourceNotFoundError()
+    }
+
+    const passwordMatched = await compare(password, user.password_hash)
+
+    // if (!passwordMatched) {
+    //   throw new ResourceNotFoundError()
+    // }
+
+    if (password !== user.password_hash) {
+      throw new ResourceNotFoundError()
+    }
```

Comment:
You should compare if the password matches using bcryptjs. 

```
  const passwordMatched = await compare(password, user.password_hash)
     if (!passwordMatched ) {
      throw new InvalidCredentialsError()
    }
    return { user }
```

Also, I think it would be better if you created another type of error, so the front-end could handle those errors differently.

Reply:
the new error class was created, but I did not change the way the comparison is made, as we are using the in-memory database, and it would break the application, as the password is not created there with bcryptjs

Reply:
You will just use the inMemory repository in your tests. Additionally, in your tests, you can hash the password before creating a user. Therefore, when you try to log in by calling the AuthUseCase, you will pass the unencrypted password, and the use case will take care of this logic for you.

Reply:
Check this line in your code -> https://github.com/MatheusSanchez/orange-back/pull/20/files#diff-02c4165bb4d6056fdbd0091c3a50e49283f76dbb73568c396713e

You already doing that 🦖

Reply:
I understand, in the tests the generated password is already being done with the hash and the comparison too:

`
const passwordMatched = await compare(password, newUser.password_hash)
expect(passwordMatched).toEqual(true)
`

The branch I was working on didn't yet have the useCase for creating a user, so I'm going to change the password comparison method in the authentication useCase

Reply:
Well noticed, can you merge that with the master branch ?

Reply:
I applied the change to the password comparison method, I couldn't see what could be changed in the tests (they all continue to pass)

As soon as I approve this PR, I will correct the merge conflicts in PR #21 and after that, I believe that the deploy is already working as in the video I sent on discord